### DISCUSSION TITLE
Unable to git pull

### DISCUSSION THREAD

**Discussion by nathondalton** [comfyanonymous/ComfyUI#4117] [General]
I cloned the repo from below. My installation works great. However, I have had a persistent issue where ComfyUI refuses to update. All plugins will update successfully, however. The error I get is, "fatal: not a git repository (or any of the parent directories): .git" I've tried wiping and re-cloning, but no dice. Any help would be appreciated. Thanks!

https://github.com/comfyanonymous/ComfyUI.git

**Comment by ltdrdata** [comment]
Does the `ComfyUI/.git` path exist? What screen appears when you run `git status` in the `ComfyUI` directory?

**Comment by nathondalton** [comment]
It did not. I did a "git init" and now it does. Lots of untracked files present. However, it still won't update via Manager. It now gives the following error, "[ComfyUI-Manager] There is no tracking branch (master) ComfyUI update fail: Remote named 'origin' didn't exist" I'm wondering why all of this horsing around when a simple git clone was supposed to be all that was needed. I'm wondering now if I need to actually pull the master branch once.

Sorry, meant to reply.
https://github.com/comfyanonymous/ComfyUI/discussions/4117#discussioncomment-10162790

**Reply by ltdrdata** [reply]
You should not execute `git init`.
You have to install ComfyUI via `git clone https://github.com/comfyanonymous/ComfyUI`

**Reply by nathondalton** [reply]
I've done that twice and I end up with this issue with a fresh install both times. It won't update, giving me the original error. For info, the install is just sitting in my Downloads folder if that makes a difference.

**Reply by ltdrdata** [reply]
It would be helpful in identifying any issues if you could accurately describe the entire process of how you installed ComfyUI, ComfyUI-Manager, and custom nodes. 

Please specify which paths you used, which commands you executed, and how you performed each step of the installation.

Your current explanation has gaps, so it's not possible to properly understand how you performed the installation.

**Reply by nathondalton** [reply]
I opened a command line, CD'd to my Downloads folder, mkdir'd a ComfyUI folder and git clone'd the git repo (first link). For ComfyUI-Manager, I used "Installation[method1] (General installation method: ComfyUI-Manager only)" at the second link below. I guess I installed ComfyUI, then I installed every plugin or product seperately. I'm new to ComfyUI and Automatic 1111 and initially watched some videos before following the documentation. Each attempt at resolving this issue was a clean install from scratch using the method stated above though. It's quite possible I made a mistake during or in the sequence of the installs (and would be happy to learn what it was).

https://github.com/comfyanonymous/ComfyUI.git
https://github.com/ltdrdata/ComfyUI-Manager

**Comment by nathondalton** [comment]
FYI, I wound up solving this by following the installation method outlined below.

ComfyUI Manager Installation Instructions - Installation Method 3
https://github.com/ltdrdata/ComfyUI-Manager?tab=readme-ov-file#installationmethod3-installation-through-comfy-cli-install-comfyui-and-comfyui-manager-at-once
Installation[method3] (Installation through comfy-cli: install ComfyUI and ComfyUI-Manager at once.)

Note, one difference here is that this method installs it to the Documents folder whereas I had ComfyUI previously in the Downloads folder. Not sure that makes a difference to git, but thought I'd throw that out there.

**Reply by ltdrdata** [reply]
> FYI, I wound up solving this by following the installation method outlined below.
> 
> ComfyUI Manager Installation Instructions - Installation Method 3 https://github.com/ltdrdata/ComfyUI-Manager?tab=readme-ov-file#installationmethod3-installation-through-comfy-cli-install-comfyui-and-comfyui-manager-at-once Installation[method3] (Installation through comfy-cli: install ComfyUI and ComfyUI-Manager at once.)
> 
> Note, one difference here is that this method installs it to the Documents folder whereas I had ComfyUI previously in the Downloads folder. Not sure that makes a difference to git, but thought I'd throw that out there.

If you are installing on Windows, it is recommended to install in a short path such as `C:\ComfyUI` if possible. Otherwise, certain files within custom nodes may not download properly due to Windows' inability to handle long paths correctly.
